_posts/2017-03-06-gcp-dataflow-streaming-example-project-0.1.0-released.md

```diff
@@ -0,0 +1,238 @@
+
+- Cloud Dataflow
+- Cloud Pub/Sub
+- Cloud Storage
+- Cloud Bigtable
+
+After enabling these services, authenticate with your GCP account by running the following commands:
+
+{% highlight bash %}
+guest$ gcloud auth login
+guest$ gcloud auth application-default login
+guest$ gcloud config set project <your-project-id>
+{% endhighlight %}
+
+These will prompt links you'll have to follow on your host OS, which will allow you
+to use your browser to authenticate.
+
+<h3>Step 3: Create your Pub/Sub topic</h3>
+
+We're going to set up the Pub/Sub topic. Your first step is to create a topic and verify that it was successful. Use the following command to create a stream named "my-stream":
+
+{% highlight bash %}
+$ inv create_pubsub_topic --topic-name=test-topic
+{% endhighlight %}
+
+<h3>Step 4: Create a Bigtable table for storing our aggregates</h3>
+
+We're using "test-table" as the table name, "test-instance" as the instance name and "cf1" as the column family. Bigtable is a NoSQL database that runs on a cluster - the instance. It has the concept of column families: groups of columns that are related and as such, are likely to be accessed roughly at the same time. In the context of this project, all the columns will lie in the same column family.
+
+Invoke the creation of the table with:
+
+{% highlight bash %}
+$ inv create_bigtable_table --instance-id=test-instance --region=us-west1-a --table-name=test-table --column-family=cf1
+{% endhighlight %}
+
+<h3>Step 5: Create a staging location for Dataflow</h3>
+
+Dataflow needs a place to store the .jars it will use to run our job. This is called the staging location.
+To create one for your project, go to https://console.cloud.google.com/storage/browser?project=YOUR-PROJECT-ID
+
+Click "Create Bucket". Fill in the appropriate details. As this is an example project, we suggest you pick the cheapest option.
+
+After creating your bucket, you might want to create a folder inside it, to serve as your staging location. You can also just use the bucket's root. To create a folder in your bucket, after you created the bucket, select it in the list and then click "Create Folder".
+
+Your staging location will be:
+
+{% highlight bash %}
+ gs://your-bucket-name/the-folder-that-you-might-have-created
+{% endhighlight %}
+
+![bucket1] [bucket1.png]
+
+![bucket2] [bucket2.png]
+
+<h3>Step 6: Submit your application to Dataflow</h3>
+
+Before you can run your application, you'll need to setup a proper config file.
+There's an example in [config/config.hocon.sample][config]. If you've been using the same names as we did, you'll only need to perform some minimal changes to that example file (namely: the project ID; adjusting the Pub/Sub topic full name; setting Dataflow's correct staging location)
+
+Then, inside your Vagrant box run:
+
+{% highlight bash %}
+guest$ inv run_project --config=path/to/config/file --fat-jar-path=path/to/fat.jar
```

Comment:
It has been assembled with
```
$ inv build_project
```
right?

But I will still add that it can be downloaded from Bintray